crates/libeq_wld/src/parser/strings.rs

```diff
@@ -39,7 +38,7 @@ pub fn decode_string(encoded_data: &[u8]) -> String {
         .map(|(encoded_char, key_char)| encoded_char ^ key_char)
         .collect();
     let (cow, _, _) = WINDOWS_1252.decode(&data);
-    cow.into_owned().trim_end_matches('\u{0}').to_string()
```

Comment:
Some files seem to have string sections that end with a bunch of null characters. In order to match the original file (and offset references in that file) when re-serializing we have to keep these around. This just means nulls may need to be stripped out later when using the strings.